src/plugins/auto/src/plugin.cpp

```diff
@@ -463,9 +502,21 @@ std::shared_ptr<ov::ICompiledModel> Plugin::compile_model_impl(const std::string
         LOG_INFO_TAG("underlying hardware does not support hardware context");
     }
     if (is_cumulative) {
-        impl = std::make_shared<AutoCumuCompiledModel>(cloned_model, shared_from_this(), device_context, auto_s_context, scheduler);
+        impl = std::make_shared<AutoCumuCompiledModel>(auto_s_context->m_model,
+                                                       shared_from_this(),
+                                                       device_context,
+                                                       auto_s_context,
+                                                       scheduler);
     } else {
-        impl = std::make_shared<AutoCompiledModel>(cloned_model, shared_from_this(), device_context, auto_s_context, scheduler);
+        if (std::static_pointer_cast<AutoSchedule>(scheduler)->m_compile_context[ACTUALDEVICE].m_is_already) {
+            // release all the models here if actual device finish compiling model.
```

Comment:
when can this happen? and if model released here,  can line#515 work?

Reply:
This model object will be used only for querying the property `ov::optimal_number_of_infer_requests`, while it is unnecessary if compilation on ACTUAL DEVICE has already finished.